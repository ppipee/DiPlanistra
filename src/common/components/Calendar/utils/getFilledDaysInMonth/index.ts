import { CalendarDisplayStateProps } from 'common/components/Calendar/types'

import getDaysInMonth from '../getDaysInMonth'
import getLastMonthDaysCalendar from '../getLastMonthDaysCalendar'
import getNextMonthDaysCalendar from '../getNextMonthDaysCalendar'

export default function getFilledDaysInMonth(displayState: CalendarDisplayStateProps) {
	const { year, month } = displayState

	const lastMonthDays = getLastMonthDaysCalendar(displayState).map(
		(dateNumber) => new Date(year, month - 1, dateNumber),
	)

	const daysInMonth = getDaysInMonth(displayState).map((dateNumber) => new Date(year, month, dateNumber))

	const nextMonthDays = getNextMonthDaysCalendar(displayState).map(
		(dateNumber) => new Date(year, month + 1, dateNumber),
	)

	return [...lastMonthDays, ...daysInMonth, ...nextMonthDays]
}
