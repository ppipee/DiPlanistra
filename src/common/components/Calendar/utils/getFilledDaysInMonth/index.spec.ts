import getFilledDaysInMonth from '.'

describe('getFilledDaysInMonth', () => {
	it('should return filled week in month', () => {
		const dates = getFilledDaysInMonth({ year: 2020, month: 11 })
		expect(dates).toMatchSnapshot()
	})

	it('should return fitted week in month', () => {
		const dates = getFilledDaysInMonth({ year: 2015, month: 1 })
		expect(dates).toMatchSnapshot()
	})
})
