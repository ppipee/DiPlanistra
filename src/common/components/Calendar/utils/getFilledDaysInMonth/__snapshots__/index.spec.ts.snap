// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFilledDaysInMonth should return filled week in month 1`] = `
Array [
  2020-11-28T17:00:00.000Z,
  2020-11-29T17:00:00.000Z,
  2020-11-30T17:00:00.000Z,
  2020-12-01T17:00:00.000Z,
  2020-12-02T17:00:00.000Z,
  2020-12-03T17:00:00.000Z,
  2020-12-04T17:00:00.000Z,
  2020-12-05T17:00:00.000Z,
  2020-12-06T17:00:00.000Z,
  2020-12-07T17:00:00.000Z,
  2020-12-08T17:00:00.000Z,
  2020-12-09T17:00:00.000Z,
  2020-12-10T17:00:00.000Z,
  2020-12-11T17:00:00.000Z,
  2020-12-12T17:00:00.000Z,
  2020-12-13T17:00:00.000Z,
  2020-12-14T17:00:00.000Z,
  2020-12-15T17:00:00.000Z,
  2020-12-16T17:00:00.000Z,
  2020-12-17T17:00:00.000Z,
  2020-12-18T17:00:00.000Z,
  2020-12-19T17:00:00.000Z,
  2020-12-20T17:00:00.000Z,
  2020-12-21T17:00:00.000Z,
  2020-12-22T17:00:00.000Z,
  2020-12-23T17:00:00.000Z,
  2020-12-24T17:00:00.000Z,
  2020-12-25T17:00:00.000Z,
  2020-12-26T17:00:00.000Z,
  2020-12-27T17:00:00.000Z,
  2020-12-28T17:00:00.000Z,
  2020-12-29T17:00:00.000Z,
  2020-12-30T17:00:00.000Z,
  2020-12-31T17:00:00.000Z,
  2021-01-01T17:00:00.000Z,
]
`;

exports[`getFilledDaysInMonth should return fitted week in month 1`] = `
Array [
  2015-01-31T17:00:00.000Z,
  2015-02-01T17:00:00.000Z,
  2015-02-02T17:00:00.000Z,
  2015-02-03T17:00:00.000Z,
  2015-02-04T17:00:00.000Z,
  2015-02-05T17:00:00.000Z,
  2015-02-06T17:00:00.000Z,
  2015-02-07T17:00:00.000Z,
  2015-02-08T17:00:00.000Z,
  2015-02-09T17:00:00.000Z,
  2015-02-10T17:00:00.000Z,
  2015-02-11T17:00:00.000Z,
  2015-02-12T17:00:00.000Z,
  2015-02-13T17:00:00.000Z,
  2015-02-14T17:00:00.000Z,
  2015-02-15T17:00:00.000Z,
  2015-02-16T17:00:00.000Z,
  2015-02-17T17:00:00.000Z,
  2015-02-18T17:00:00.000Z,
  2015-02-19T17:00:00.000Z,
  2015-02-20T17:00:00.000Z,
  2015-02-21T17:00:00.000Z,
  2015-02-22T17:00:00.000Z,
  2015-02-23T17:00:00.000Z,
  2015-02-24T17:00:00.000Z,
  2015-02-25T17:00:00.000Z,
  2015-02-26T17:00:00.000Z,
  2015-02-27T17:00:00.000Z,
]
`;
